### PR TITLE
Button Functionality and Menu Transitions Complete. Options Menu Set …

### PR DESCRIPTION
**Pull request by SeanAbnerNash**
…up and ready for script implementation

### CODE REVIEW

File: Assets/Scripts/UIController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class UIController : MonoBehaviour
+{
+    // Start is called before the first frame update
+    private bool m_menuInView = true;
+    private bool m_optionsInView = false;
+    void Start()
```
Comment (CathalRedmond) [inline Assets/Scripts/UIController.cs]: If it's not being used you could get rid of it.

File: Assets/Scripts/UIController.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        setTransitions();
+   
+    }
+
+    public void exitOptionButtonPressed()
+    {
+        m_menuInView = true;
+        m_optionsInView = false;
+        setTransitions();
+        Debug.Log("HERE");
```
Comment (CathalRedmond) [inline Assets/Scripts/UIController.cs]: Don't need this now that you're finished

File: Assets/Scripts/buttonAnimation.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems; // 1
```
Comment (CathalRedmond) [inline Assets/Scripts/buttonAnimation.cs]: What is with the //1 comment?

File: Assets/Scripts/optionTransitionControl.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections;
```
Comment (CathalRedmond) [inline Assets/Scripts/optionTransitionControl.cs:1]: This Script in general seems to copy a lot of code from the menuTransitionControl is there maybe a way to have the code in only one script?

File: Assets/Scripts/optionButton.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class optionButton : MonoBehaviour
+{
+    private UIController script;
```
Comment (CathalRedmond) [inline Assets/Scripts/optionButton.cs]: script should be m_script

**Review by CathalRedmond — CHANGES_REQUESTED** [review]
In general,
- You have a lot of empty lines, they should be gotten rid of
- Functions in C# should start with a capital letter
- You also seem to not have consistent spacing and line indentation in some of the files
- You also have empty functions that do nothing like the default start and update functions they need to be gotten rid of

File: Assets/Scripts/optionTransitionControl.cs (hunk 4 above)
Comment (SeanAbnerNash) [reply Assets/Scripts/optionTransitionControl.cs:1]: Not from the way things are currently structured. I want each menu to be able to move independently of the other. Having two scripts means the code needed is simpler.

Comment (CathalRedmond) [reply Assets/Scripts/optionTransitionControl.cs:1]: Ok Just Checking

**Pull request merged**